File: BookingApp/src/main/java/com/example/BookingApp/reservations/model/Action.java
package com.example.BookingApp.reservations.model;

import com.example.BookingApp.renting.model.RentingItem;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name="actions")
public class Action {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private RentingItem rentingItem;
    private Date startTime;
    private Date endTime;
    private Double price;
    private boolean reserved;


}

File: BookingApp/frontend/src/app/shared/models/reservations/Action.ts
import { RentingItem } from "./RentingItem";
import { ReservedRentingItem } from "./ReservedRentingItem";

export class Action{
    public id: Number;
    public rentingItemId: Number;
    public startTime: Date;
    public endTime: Date;
    public capacity : Number;
    public additionalServices : String;
    public price : Number;
    public reserved : Boolean;
    public rentingItem : RentingItem;
    public reservedRentingItem : ReservedRentingItem;

    constructor(id : Number, rentingItemId : Number,startTime : Date,endTime : Date,capacity: Number,
        additionalServices : String, price : Number, reserved : Boolean, rentingItem : RentingItem){
            this.id = id;
            this.rentingItemId = rentingItemId;
            this.startTime = startTime;
            this.endTime = endTime;
            this.capacity = capacity;
            this.additionalServices = additionalServices;
            this.price = price;
            this.reserved = reserved;
            this.rentingItem = rentingItem;
            
        }
}

